### ISSUE TITLE
Question on experiment setting

### ISSUE THREAD

**Issue by Weiting-Gao** [successar/FRESH#6]
Thanks for the great work!
I have a question regarding the experiment setting. As pred is trained and tested only on text provided by ext, does it mean only the original test data is fed into pred for both training and testing (since supp and ext only extract rationales for original test data)? By original test data, I mean the test set of supp.

**Comment by successar** [comment]
Test data is never fed into the pred model while training. The way we train it is 1) Train the supp model training data and then extract rationales for each training data example and 2) Use these rationales as input to pred model and train to predict the final prediction.

At test time, we simply use the supp+ext to extract the rationale for that test point and then feed it to the pred model to get the final prediction.

**Comment by Weiting-Gao** [comment]
Thanks for your reply, it's really helpful!

I got another question about the model. I wonder are supp and pred should be the same black box model? Your code seems they are same, but in your paper talk, you use 'black box model' for saliency scorer, 'black box model 2' for classifier, it seems they are different.

**Comment by successar** [comment]
You can use any classification model for supp and pred and they can be different. They can also be of same *type* but have different parameters (in our case, we use BERT for both supp and pred but they don't share parameters; the BERT trained for supp is independent of the BERT trained for pred). Hence we use "black box model" and "black box model 2" in the talk, since they are two different models.

**Comment by Weiting-Gao** [comment]
Got it, thanks a lot!
